File: app/src/main/java/com/codekk/mvp/presenter/RecommendListPresenterImpl.java
package com.codekk.mvp.presenter;

import com.codekk.mvp.model.RecommendListModel;
import com.codekk.mvp.view.ViewManager;
import com.codekk.net.Api;
import com.codekk.net.NetFunc;
import com.codekk.ui.base.BasePresenterImpl;
import com.common.widget.StatusLayout;

import io.reactivex.network.manager.RxNetWork;

/**
 * by y on 2017/5/20.
 */

public class RecommendListPresenterImpl extends BasePresenterImpl<ViewManager.RecommendListView, RecommendListModel> implements PresenterManager.RecommendListPresenter {

    public RecommendListPresenterImpl(ViewManager.RecommendListView view) {
        super(view);
    }

    @Override
    public void netWorkRequest(int page) {
        netWork(RxNetWork
                .observable(Api.RecommendService.class)
                .getRecommendList(page)
                .map(new NetFunc<>()));
    }

    @Override
    public void onNetWorkSuccess(RecommendListModel data) {
        if (data.getRecommendArray().isEmpty()) {
            view.noMore();
            setRootViewState(StatusLayout.EMPTY);
        } else {
            super.onNetWorkSuccess(data);
        }
    }
}

File: app/src/main/java/com/codekk/mvp/presenter/RecommendListPresenterImpl.kt
package com.codekk.mvp.presenter

import com.codekk.Api
import com.codekk.NetFunc
import com.codekk.RecommendService
import com.codekk.mvp.model.RecommendListModel
import com.codekk.mvp.view.ViewManager
import com.codekk.ui.base.BasePresenterImpl

import io.reactivex.network.RxNetWork


/**
 * by y on 2017/5/20.
 */

class RecommendListPresenterImpl(view: ViewManager.RecommendListView) : BasePresenterImpl<ViewManager.RecommendListView, RecommendListModel>(view), PresenterManager.RecommendListPresenter {

    override fun netWorkRequest(page: Int) {
        netWork(RxNetWork
                .observable(RecommendService::class.java)
                .getRecommendList(page)
                .map(NetFunc()))
    }

    override fun onNetWorkSuccess(data: RecommendListModel) {
        if (data.recommendArray.isEmpty()) {
            view?.noMore()
        } else {
            super.onNetWorkSuccess(data)
        }
    }
}
